repo.go

```diff
@@ -526,6 +534,50 @@ func (r *Repo) Sign(name string) error {
 	return r.local.SetMeta(name, b)
 }
 
+// AppendSignature allows users to append a signature generated with an external tool.
+func (r *Repo) AppendSignature(name string, signature data.Signature) error {
+	role := strings.TrimSuffix(name, ".json")
+	if !verify.ValidRole(role) {
+		return ErrInvalidRole{role}
+	}
+
+	// Check Key ID is in valid for the role.
+	db, err := r.db()
+	if err != nil {
+		return err
+	}
+	roleData := db.GetRole(role)
+	if roleData == nil {
+		return ErrInvalidRole{role}
+	}
+	if !roleData.ValidKey(signature.KeyID) {
+		return verify.ErrInvalidKey
+	}
+
+	s, err := r.SignedMeta(name)
+	if err != nil {
+		return err
+	}
+
+	// Iterate through signatures and remove a signature with a matching ID
```

Comment:
I think removing existing signatures for the keyID, and adding a new one, is a good idea, but I'm afraid the name of the function is then a bit misleading. Two choices: either we split adding/appending and updating into two functions, or we give the function a better name (like AddOrUpdateSignature).

Reply:
This is also how `Sign` works! Signing twice over the same manifest with the same key replaces the old one. On the other hand, it's the same to me to skip appending if already present. Would that make sense so it doesn't update?